# <center> <b> Hypothesis testing  </b> </center>


####  Hypothesis testing is a statistical technique employed to draw conclusions about population characteristics using sample data. By comparing the observed data with what's expected under a null hypothesis, which embodies the status quo or a default assumption, researchers aim to ascertain whether the observed outcomes are statistically significant enough to refute the null hypothesis in favor of an alternative hypothesis. This method allows for the assessment of the credibility of assertions or claims based on the evidence provided by the sample data, facilitating informed decision-making in various fields by providing a structured framework for drawing meaningful conclusions.

#### Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.

In [1]:
#Import norm module from the scipy.stats library
from scipy.stats import norm

In [2]:
# Given data
population_std = 2.3
sample_size = 1000
sample_mean = 11.5
population_mean = 11
alpha = 0.05

## 1.Formulate Hypotheses:
#### Null Hypothesis (H0): μ ≤ 11 minutes

#### Alternative Hypothesis (Ha): μ > 11 minutes

## 2.Choose Test Statistic:
##### Since the population standard deviation is known and the sample size is large (n > 30), we use the Z-test.

## 3.Calculate Z-score:

### Z = (x̄ - μ) / (σ / √n)
##### sample_mean(x̄)=11.5 minutes
##### population_mean(μ)=11 minutes
##### population_std(σ)=2.3 minutes
##### sample_size(n)=1000
##### significance level(α)=0.05

In [3]:
# Calculate Z-score
Z = (sample_mean - population_mean) / (population_std / (sample_size ** 0.5))
Z

6.874516652539955

## 4.Determine Critical Value:
##### At α=0.05 the critical Z-value for a one-tailed test is approximately 1.645.

In [4]:
# Find critical value for one-tailed test
critical_value = norm.ppf(1-alpha)
critical_value

1.6448536269514722

## 5.Make Decision:
##### Since the calculated Z-value (6.89) is much greater than the critical Z-value (1.645), we fail to accept null hypothesis.

## 6.Interpret the Results:

Based on the decision, interpret whether there is sufficient evidence to support the psychologist's claim.

## 7.Conclusion:
##### There is enough evidence to conclude that the average time working mothers spend talking to their children is more than 11 minutes per day.

#### Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

In [5]:
#Import t module from the scipy.stats library
from scipy.stats import t

In [6]:
# Given data
sample_size = 40
sample_mean = 4.6
sample_std = 0.8
population_mean = 5
alpha = 0.05

## 1.Formulate Hypotheses:
#### Null Hypothesis (H0): μ ≥ 5  minutes
#### Alternative Hypothesis (Ha):  μ < 5  minutes

## 2.Choose Test Statistic:
##### If the population standard deviation is not given, and the sample size is less than 30, we use the T-test. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

## 3.Calculate t-score:
### t = (x̄ - μ) / (s / √n)
##### Sample_mean(x̄)=4.6 minutes
##### Population_mean(μ)=5 minutes
##### Sample_std(s)=0.8 minutes
##### Sample_size(n)=40
##### Significance level(α)=0.05
##### Degree of freedom(n-1)=39

In [7]:
# Calculate t-score
t_score = (sample_mean - population_mean) / (sample_std / (sample_size ** 0.5))
t_score 

-3.162277660168382

## 4.Determine Critical Value:
##### At α=0.05 and degrees of freedom (df) = 39, the critical t-value for a one-tailed test is approximately -1.685.

In [8]:
# Degrees of freedom
df = sample_size - 1
df 

39

In [9]:
# Find critical value for one-tailed test
critical_value = t.ppf(alpha, df)
critical_value

-1.6848751194974

## 5.Make Decision:
##### Since the calculated t-value (-3.16) is less than the critical t-value (-1.685), we fail to accept the null hypothesis.

## 6.Interpret the Results:

##### Based on the decision, interpret whether there is sufficient evidence to support the coffee shop's claim.

## 7.Conclusion:
##### There is enough evidence to conclude that the average wait time for customers at the coffee shop is less than 5 minutes.